### ISSUE TITLE
[tune] stopping before stop condition is true

### ISSUE THREAD

**Issue by himat** [ray-project/ray#4224]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04.6 LTS (GNU/Linux 4.15.0-43-generic x86_64)
- **Ray installed from (source or binary)**: binary (pip)
- **Ray version**: 0.6.3
- **Python version**: 3.6.4
- **Exact command to reproduce**:

### Describe the problem

For some reason, Tune is stopping trials before they've reached the set stop condition of 60% avg_val_acc that I set.
Is this expected behavior?

### Source code / logs

```
2019-03-02 14:45:41,458 WARNING function_runner.py:54 -- Trial detected as completed; re-reporting last result. To avoid this, include done=True upon the las
t reporter call.
Result for tune_train_0_batch_size=8,lr=5e-05:
  avg_val_acc: 53.84615384615385
  date: 2019-03-02_14-45-41
  done: true
  experiment_id: 013cd743e0aa43728b70f17f4c4c928e
  hostname: marten
  iterations_since_restore: 41
  node_ip: some_ip
  pid: 29474
  time_since_restore: 663.6665663719177
  time_this_iter_s: 7.0081467628479
  time_total_s: 663.6665663719177
  timestamp: 1551555941
  timesteps_since_restore: 0
  training_iteration: 41

```

But my stop condition is that `avg_val_acc` should be 60, but as you can see it only reached 53% above:
```
 ray.init()

 num_gpu = 1
 num_cpu = num_gpu * 8

 all_trials = tune.run_experiments({
     exp_name: {
         "run": train_func,
         "stop": {"avg_val_acc": 60},
         "local_dir": "./tune_checkpoints",
         "config": {
                 "lr": tune.grid_search([5e-5, 0.1]),
                 "batch_size": tune.grid_search([8, 192]),
             },
         "resources_per_trial": {
             "cpu": num_cpu, "gpu": num_gpu
             },

         }

     })
```

Or is there some default max num iterations that's being reached, since I noticed that all of my experiments stopped after 40 iterations. How would I change this for the default scheduler?
```
== Status ==
Using FIFO scheduling algorithm.
Resources requested: 0/32 CPUs, 0/2 GPUs
Memory usage on this node: 6.1/202.7 GB
Result logdir: ./tune_checkpoints/tune_clf_pt_pacs_OnArt_ValStopping
TERMINATED trials:
 - tune_train_0_batch_size=8,lr=5e-05:  TERMINATED [pid=29474], 663 s, 41 iter
 - tune_train_1_batch_size=192,lr=5e-05:        TERMINATED [pid=29475], 437 s, 41 iter
 - tune_train_2_batch_size=8,lr=0.1:    TERMINATED [pid=30610], 828 s, 41 iter
 - tune_train_3_batch_size=192,lr=0.1:  TERMINATED [pid=30620], 615 s, 41 iter


```

**Comment by richardliaw** [comment]
Hm, it looks like the function `train_func` is finishing before the val_acc reaches a higher score; is there anything happening there?

**Comment by himat** [comment]
I think it might have been that 40 was the number of epochs I had told the training loop to run for actually

**Comment by richardliaw** [comment]
OK; sounds like this is resolved, so I'll close this for now (and feel free to reopen or open a new issue if something else comes up).
